components/form/index.tsx

```diff
@@ -59,9 +59,15 @@ export default function StoryForm() {
       onSubmit={async (values: LoginFormInputs, actions) => {
         try {
           actions.setSubmitting(true)
+          const anonymousValue = values.anonymous
+          const payload = {
+            ...values,
+            // cast the string anonymous value to boolean
+            anonymous: JSON.parse(anonymousValue),
+          }
```

Comment:
Less complex solve - I leave the value as a string until submission when it gets cast to a boolean. `yup.boolean` must be parsing 'true' as valid? 🤷

Reply:
Geez yup, get it together.